### ISSUE TITLE
Deprecate $resourceProvider.defaults.stripTrailingSlashes

### ISSUE THREAD

**Issue by mgol** [angular/angular.js#15159]
***Note*: This issue is blocked on #15158.**

***Note*: for support questions, please use one of these channels: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#question. This repository's issues are reserved for feature requests and bug reports.**

**Do you want to request a *feature* or report a *bug*?**
A refactor.

**What is the current behavior?**
`$resourceProvider.defaults.stripTrailingSlashes` may have a `true` value (which is now a default, to be changed in #15158).

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://plnkr.co or similar (template: http://plnkr.co/edit/tpl:yBpEi4).**
N/A

**What is the expected behavior?**
`$resourceProvider.defaults.stripTrailingSlashes` shouldn't exist. People should pass to `$resource` URLs in the form the backend expects them instead of relying on Angular to convert them.

**What is the motivation / use case for changing the behavior?**
The flag was only added because `$resource` was stripping trailing slashes and a lot of people (including me) were complaining they can't use `$resource` with their backend if it used URLs with trailing slashes (one example being Django). We don't have a flag to ensure a trailing slash, only to remove one - if we considered this API important, for consistency we'd have to add the other one as well. This doesn't seem worth it; there's no big gain in passing URLs different to the ones recognized by the backend to `$resource`.

**Which versions of Angular, and which browser / OS are affected by this issue? Did this work in previous versions of Angular? Please also test with the latest stable and snapshot (https://code.angularjs.org/snapshot/) versions.**
N/A

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**
N/A

**Comment by mgol** [comment]
The intention behind this issue was that people can provide URLs exactly as the backend recognizes them so the switch can be switched to not do any stripping and deprecated/removed afterwards. What I didn't think of are parametrized URLs where you don't have to provide all the parameters. For example, take:
```
/Path/:a/:b/
```
If you don't want a trailing slash, you can just remove it:
```
/Path/:a/:b
```
and it'll work as desired if you pass both `a` & `b`. What if you don't pass `b` but only `a: 1`? With `disableTrailingSlashes` you'll end up with:
```
/Path/1/
```
which may not be desired. Hence, the `stripTrailingSlashes` option is still needed.

The issue mentioned we don't have an option to ensure trailing slashes but it's not needed. If your backend requires them, you just need to set `stripTrailingSlashes` to false and use trailing slashes in your URLs; the URL:
```
/Path/:a/:b/
```
will produce the following URLs:
```
/Path/1/2/ # a = 1, b = 2
/Path/1/ # a = 1, b not provided
```

Hence, I'm closing this issue as "Won't fix".
